core/src/main/java/co/cask/wrangler/steps/parser/JsParser.java

```diff
@@ -90,7 +90,7 @@ public JsParser(int lineno, String detail, String col, int maxDepth) {
           JsonElement element = null;
           if(value instanceof String) {
             String document = (String) value;
-            element = parser.parse(document);
+            element = parser.parse(document.trim());
```

Comment:
When treating a `json` file as a `DataType.TEXT`, extra spaces are consistently added at the end of the input string, so the `JsonParser` here throws an error, unless you trim the input string before parsing it.